### ISSUE TITLE
Android cannot reset app

### ISSUE THREAD

**Issue by nabeel-ibrahim** [appium/appium#5885]
using driver.resetApp() throws an error. I am using appium 1.4.16 and testing on the genymotion emulator. Is there any other way to reset the app?
```org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 39 milliseconds
Build info: version: '2.48.2', revision: '41bccdd', time: '2015-10-09 19:55:52'
System info: host: 'nibrahim-mbp.local', ip: '192.168.56.1', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.11.1', java.version: '1.8.0_66'
Driver info: io.appium.java_client.android.AndroidDriver
Capabilities [{app=/Users/nibrahim/Desktop/Automation/QA_Automation/Android/theScoreApp-google-debug.apk, appPackage=com.fivemobile.thescore.debug, networkConnectionEnabled=true, warnings={}, databaseEnabled=false, deviceName=192.168.56.101:5555, platform=LINUX, desired={app=/Users/nibrahim/Desktop/Automation/QA_Automation/Android/theScoreApp-google-debug.apk, appPackage=com.fivemobile.thescore.debug, platformVersion=4.4.4, udid=192.168.56.101:5555, platformName=Android, deviceName=generic}, platformVersion=4.4.4, webStorageEnabled=false, locationContextEnabled=false, browserName=Android, takesScreenshot=true, javascriptEnabled=true, udid=192.168.56.101:5555, platformName=Android}]
Session ID: fa23a990-35f3-4e53-ac5a-24c1c757571a```

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:647)
	at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:42)
	at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:189)
	at io.appium.java_client.AppiumDriver.resetApp(AppiumDriver.java:202)

**Comment by IntelloScooby** [comment]
I am also facing this issue but it is very random. Sometimes the same test passes and resetApp works perfectly fine and at times the same code fails with the error "An unknown server-side error occurred while processing the command.  (WARNING: The server did not provide any stacktrace information)"

**Comment by imurchie** [comment]
Can you provide the Appium server logs so we can try to diagnose the issue?

**Comment by IntelloScooby** [comment]
Since this is random issue, I do not have the appium server log right now but I will immediately update this thread when I receive the error again.

**Comment by IntelloScooby** [comment]
I was able to resolve this issue. In my code, I was closing and immediately launching the app 2 times when running on android emulator and in the @After scenario I was resetting the App. Once I introduced a sleep of 4 seconds between closeApp and launchApp, the issue was resolved and it did not break on resetApp.

I am not aware about the reason why it is behaving this way but these steps resolved the issue for me.

**Comment by imurchie** [comment]
Weird. But I'm glad things worked out!

**Comment by deepaksporty** [comment]
I am too getting the same weird behavior. resetApp works fine sometimes and throws exception other times. I am getting the same error as mentioned above and tried the workaround tried by IntelloScooby

"I was able to resolve this issue. In my code, I was closing and immediately launching the app 2 times when running on android emulator and in the @After scenario I was resetting the App. Once I introduced a sleep of 4 seconds between closeApp and launchApp, the issue was resolved and it did not break on resetApp.
I am not aware about the reason why it is behaving this way but these steps resolved the issue for me."

I am using 1.4.16.1 version of appium.

Kindly help
